x-pack/plugins/security_solution/server/usage/detections/ml_jobs/get_metrics.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  KibanaRequest,
+  SavedObjectsClientContract,
+  Logger,
+} from '../../../../../../../src/core/server';
+import type { MlDatafeedStats, MlJob, MlPluginSetup } from '../../../../../ml/server';
+import type { MlJobMetric, MlJobUsageMetric } from './types';
+
+import { isJobStarted } from '../../../../common/machine_learning/helpers';
+import { isSecurityJob } from '../../../../common/machine_learning/is_security_job';
+import { getInitialMlJobUsage } from './get_initial_usage';
+import { updateMlJobUsage } from './update_usage';
+import { getJobCorrelations } from './transform_utils/get_job_correlations';
+
+export interface GetMlJobMetricsOptions {
+  mlClient: MlPluginSetup | undefined;
+  savedObjectsClient: SavedObjectsClientContract;
+  logger: Logger;
+}
+
+export const getMlJobMetrics = async ({
+  mlClient,
+  savedObjectsClient,
+  logger,
+}: GetMlJobMetricsOptions): Promise<MlJobUsageMetric> => {
+  let jobsUsage = getInitialMlJobUsage();
+
+  if (mlClient == null) {
+    logger.debug(
+      'Machine learning client is null/undefined, therefore not collecting telemetry from it'
+    );
+    // early return if we don't have ml client
+    return {
+      ml_job_usage: getInitialMlJobUsage(),
+      ml_job_metrics: [],
+    };
+  }
+
+  try {
+    const fakeRequest = { headers: {} } as KibanaRequest;
+
+    const modules = await mlClient.modulesProvider(fakeRequest, savedObjectsClient).listModules();
+    const moduleJobs = modules.flatMap((module) => module.jobs);
+    const jobs = await mlClient.jobServiceProvider(fakeRequest, savedObjectsClient).jobsSummary();
+
+    jobsUsage = jobs.filter(isSecurityJob).reduce((usage, job) => {
```

Comment:
One interesting thing I noticed last week is that this doesn't represent custom security jobs accurately. This is because `updateMlJobUsage` checks for if a job has been tagged with `siem` or `security`, but it is pretty unlikely that a user is going to create a custom job and label it with our labels. We probably don't want to filter with this condition for custom ML jobs and simplify `updateMlJobUsage`, but you don't need to worry about this in this PR. Just making you aware.

Reply:
Ok, I am not going to change any of this part at this point for this PR.